### PR TITLE
feat(build-cli): Add support for selecting only changed packages

### PR DESCRIPTION
**Pull request by tylerbutler**
Adds a selection flag that can be used to select packages that have changed when compared to a base branch. The intended use case is something like this:

```shell
flub exec --changed -- "npm run test"
```

That command will run `npm run test` in all packages that have changes since `main`.

The base branch can be changed using the `--branch` flag:

```
flub exec --changed --branch next -- "npm run format"
```

Because this is a general-purpose filter, the new flag will work on all commands that use the standard filtering flags and functions.

### CODE REVIEW

File: build-tools/packages/build-cli/src/commands/list.ts [hunk 1]
```diff
@@ -103,7 +103,8 @@ export default class ListCommand extends BaseCommand<typeof ListCommand> {
 
 		const filterOptions = parsePackageFilterFlags(this.flags);
 		const packageList = await pnpmList(rgOrPackage.repoPath);
-		const filtered = filterPackages(packageList, filterOptions)
+		const filtered = (await filterPackages(packageList, filterOptions))
+			// eslint-disable-next-line unicorn/no-await-expression-member
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/commands/list.ts:107]: IMO we should disable this rule or reduce it to a warning. But I can also update it to use an intermediate variable if desired.

Comment (jason-ha) [reply build-tools/packages/build-cli/src/commands/list.ts:107]: I don't know about the rule. It probably is good for a case like this if you wanted to debug something. Having the local can be useful and doesn't impact the block adversely.

**Review by jason-ha — APPROVED** [review]
I like this feature. Thanks for putting things together.

File: build-tools/packages/build-cli/src/commands/list.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/commands/list.ts]: Updated in latest.

**Pull request merged**